### PR TITLE
Add mlir-hlo to external llvm targets

### PR DESCRIPTION
**Pull request by brianretford**
Rolls llvm forward to the version specified in the  (newly added) mlir-hlo vendor submodule.

### CODE REVIEW

File: environment.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ channels:
 dependencies:
   - ccache=4.1
   - cffi=1.14.4
-  - cmake=3.13.4
+  - cmake=3.14
```
Comment (flaub) [inline environment.yml]: This breaks openvino and other potential integaration.

File: environment.yml [hunk 2]
```diff
@@ -12,6 +12,7 @@ dependencies:
   - numpy=1.18.5
   - pip=21.0.1
   - pre_commit=2.11.1
+  - pybind11
```
Comment (flaub) [inline environment.yml]: Is this needed?

**Review by flaub — DISMISSED** [review]

File: environment.yml (hunk 1 above)
Comment (brianretford) [reply environment.yml]: reverted, not sure why I bumped it forward anymore, everything works as is

File: environment.yml (hunk 2 above)
Comment (brianretford) [reply environment.yml]: not now that I'm not building the mlir/mhlo python bindings, removed

**Review by flaub — APPROVED** [review]

**Pull request merged**